### ISSUE TITLE
Foreground objects should be rendered in front of background objects

### ISSUE THREAD

**Issue by yosun** [Lavreniuk/generative-interior-design#1]
Here is an example where I think you are prioritizing the background scenery too heavily. The door should be behind the furniture in the center. 

[![Image from Gyazo](https://i.gyazo.com/bec4d1a1719a93d60883928717ff5529.jpg)](https://gyazo.com/bec4d1a1719a93d60883928717ff5529)

**Comment by melgor** [comment]
That's true. In our pipeline all doors and windows will be fully visible, no without any furniture covering them up. We didn't have time to resolve this issue during the competition but it is severe issue.

**Comment by yosun** [comment]
Maybe do a foreground z-index check

It seems the chair is covering the window partially

**Comment by melgor** [comment]
What do you mean by foreground z-index check?
I don't know this technique and google only said sth about CSS styling

**Comment by andyhtet** [comment]
I have also thought about this issue when I test this repository. It might not be the best solution but how about we generate a rough image first, get the masked area of foreground objects in that generated image, and then refine/merge the original mask using that new mask to include the area of objects that will be generated? Then we generate the full image again using that updated mask, assuming the area of foreground objects in front of the door or windows is included in the masked inpainting.
